File: backends/java-spring/src/main/java/com/authorization/http/responses/client/ShowResponse.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.http.responses.client;

import com.authorization.domain.client.valueobjects.ClientDetailVo;
import com.authorization.support.http.responses.AbstractResponse;
import java.time.format.DateTimeFormatter;

/**
 * クライアント詳細Responseクラスです（PHP版 ShowResponse 相当）。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
public class ShowResponse extends AbstractResponse {

    private static final DateTimeFormatter FMT = DateTimeFormatter.ISO_DATE_TIME;

    private final Long id;
    private final String name;
    private final String identifier;
    private final String postCode;
    private final String pref;
    private final String city;
    private final String address;
    private final String building;
    private final String tel;
    private final String email;
    private final Integer status;
    private final String startAt;
    private final String stopAt;
    private final String createdAt;
    private final String updatedAt;

    /**
     * コンストラクタ。
     *
     * @param detail クライアント詳細ValueObject
     */
    public ShowResponse(ClientDetailVo detail) {
        this.id = detail.getId();
        this.name = detail.getName();
        this.identifier = detail.getIdentifier();
        this.postCode = detail.getPostCode();
        this.pref = detail.getPref();
        this.city = detail.getCity();
        this.address = detail.getAddress();
        this.building = detail.getBuilding();
        this.tel = detail.getTel();
        this.email = detail.getEmail();
        this.status = detail.getStatus();
        this.startAt = detail.getStartAt() != null ? detail.getStartAt().format(FMT) : "";
        this.stopAt = detail.getStopAt() != null ? detail.getStopAt().format(FMT) : "";
        this.createdAt = detail.getCreatedAt() != null ? detail.getCreatedAt().format(FMT) : "";
        this.updatedAt = detail.getUpdatedAt() != null ? detail.getUpdatedAt().format(FMT) : "";
    }
}

File: backends/csharp-aspnet/src/Authorization.Api/Http/Responses/Client/ShowResponse.cs
// This is a program developed by BobTabo.
//
// Copyright (c) 2026 BobTabo. All Rights Reserved.
using Authorization.Api.Domain.Client;
using Authorization.Api.Support;
using Authorization.Api.Support.Http.Responses;

namespace Authorization.Api.Http.Responses.Client;

/// <summary>クライアント詳細Responseクラスです（PHP版 ShowResponse 相当）。</summary>
public sealed class ShowResponse : AbstractResponse
{
    /// <summary>コンストラクタ。</summary>
    /// <param name="c">クライアント詳細</param>
    public ShowResponse(ClientDetailVo c)
    {
        Id = c.Id;
        Name = c.Name;
        Identifier = c.Identifier;
        PostCode = c.PostCode;
        Pref = c.Pref;
        City = c.City;
        Address = c.Address;
        Building = c.Building;
        Tel = c.Tel;
        Email = c.Email;
        Status = c.Status;
        StartAt = DateFormat.ToMinute(c.StartAt);
        StopAt = DateFormat.ToMinute(c.StopAt);
        CreatedAt = DateFormat.ToMinute(c.CreatedAt);
        UpdatedAt = DateFormat.ToMinute(c.UpdatedAt);
        Version = c.Version;
    }

    /// <summary>クライアントID</summary>
    public long Id { get; }

    /// <summary>名称</summary>
    public string Name { get; }

    /// <summary>識別子</summary>
    public string Identifier { get; }

    /// <summary>郵便番号</summary>
    public string PostCode { get; }

    /// <summary>都道府県</summary>
    public string Pref { get; }

    /// <summary>市区町村</summary>
    public string City { get; }

    /// <summary>番地</summary>
    public string Address { get; }

    /// <summary>建物名</summary>
    public string Building { get; }

    /// <summary>電話番号</summary>
    public string Tel { get; }

    /// <summary>メールアドレス</summary>
    public string Email { get; }

    /// <summary>ステータス</summary>
    public int Status { get; }

    /// <summary>利用開始日時（yyyy-MM-dd HH:mm形式）</summary>
    public string? StartAt { get; }

    /// <summary>利用停止日時（yyyy-MM-dd HH:mm形式）</summary>
    public string? StopAt { get; }

    /// <summary>作成日時（yyyy-MM-dd HH:mm形式）</summary>
    public string CreatedAt { get; }

    /// <summary>更新日時（yyyy-MM-dd HH:mm形式）</summary>
    public string UpdatedAt { get; }

    /// <summary>楽観排他ロック用バージョン</summary>
    public int Version { get; }
}
